# Bundesliga Home Advantage during COVID-19

## Research Question

Did the absence of spectators during the COVID-19 pandemic reduce the home advantage in the German Bundesliga?

This project analyses Bundesliga match data from multiple seasons before and during COVID-19.

## 1. Import Libaries

In [5]:
import pandas as pd
import matplotlib.pyplot as plt


Matplotlib is building the font cache; this may take a moment.


## 2. Load the Data

In this section, the Bundesliga dataset for the 2018/19 season is loaded into a pandas DataFrame.

In [8]:
df_1819 = pd.read_csv("data/D1_1819.csv")

## 3. Explore the Dataset

Before starting the analysis, it is important to understand the structure of the dataset.

In this section, we examine the first rows of the dataset, identify the available variables, and inspect the dimensions and data types. This helps to verify that the dataset has been loaded correctly and provides an overview of the available information.

In [14]:
df_1819.head()



,Div,Date,HomeTeam,AwayTeam,FTHG,FTAG,FTR,HTHG,HTAG,HTR,...,BbAv<2.5,BbAH,BbAHh,BbMxAHH,BbAvAHH,BbMxAHA,BbAvAHA,PSCH,PSCD,PSCA
0,D1,24/08/2018,Bayern Munich,Hoffenheim,3,1,H,1,0,H,...,3.55,22,-2.00,1.92,1.87,2.05,1.99,1.23,7.15,14.10
1,D1,25/08/2018,Fortuna Dusseldorf,Augsburg,1,2,A,1,0,H,...,1.76,20,0.00,1.80,1.76,2.17,2.11,2.74,3.33,2.78
2,D1,25/08/2018,Freiburg,Ein Frankfurt,0,2,A,0,1,A,...,1.69,20,-0.25,2.02,1.99,1.92,1.88,2.52,3.30,3.07
3,D1,25/08/2018,Hertha,Nurnberg,1,0,H,1,0,H,...,1.76,20,-0.25,1.78,1.74,2.21,2.14,1.79,3.61,5.21
4,D1,25/08/2018,M'gladbach,Leverkusen,2,0,H,0,0,D,...,2.32,20,0.00,2.13,2.07,1.84,1.78,2.63,3.70,2.69


The following output displays the first five matches of the dataset.

In [ ]:
df_1819.columns



61

In [31]:
df_1819.shape

(306, 61)

The dataset contains 306 matches and 61 variables, confirming that the complete Bundesliga season 2018/19 has been loaded successfully.

In [32]:
df_1819.info()

<class 'pandas.DataFrame'>
RangeIndex: 306 entries, 0 to 305
Data columns (total 61 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   Div       306 non-null    str    
 1   Date      306 non-null    str    
 2   HomeTeam  306 non-null    str    
 3   AwayTeam  306 non-null    str    
 4   FTHG      306 non-null    int64  
 5   FTAG      306 non-null    int64  
 6   FTR       306 non-null    str    
 7   HTHG      306 non-null    int64  
 8   HTAG      306 non-null    int64  
 9   HTR       306 non-null    str    
 10  HS        306 non-null    int64  
 11  AS        306 non-null    int64  
 12  HST       306 non-null    int64  
 13  AST       306 non-null    int64  
 14  HF        306 non-null    int64  
 15  AF        306 non-null    int64  
 16  HC        306 non-null    int64  
 17  AC        306 non-null    int64  
 18  HY        306 non-null    int64  
 19  AY        306 non-null    int64  
 20  HR        306 non-null    int64  
 21  AR  

## 4. Select Relevant Variables

The original dataset contains 61 variables. However, only a subset of these variables is required to analyse the home advantage in the Bundesliga.

For this project, we focus on:
- Date
- HomeTeam
- AwayTeam
- Full Time Home Goals (FTHG)
- Full Time Away Goals (FTAG)
- Full Time Result (FTR)

In [35]:
df_1819[["Date","HomeTeam","AwayTeam","FTHG","FTAG","FTR"]].head()


,Date,HomeTeam,AwayTeam,FTHG,FTAG,FTR
0,24/08/2018,Bayern Munich,Hoffenheim,3,1,H
1,25/08/2018,Fortuna Dusseldorf,Augsburg,1,2,A
2,25/08/2018,Freiburg,Ein Frankfurt,0,2,A
3,25/08/2018,Hertha,Nurnberg,1,0,H
4,25/08/2018,M'gladbach,Leverkusen,2,0,H


In [37]:
relevant_df = df_1819[
    ["Date", "HomeTeam", "AwayTeam", "FTHG", "FTAG", "FTR"]
]

In [38]:
relevant_df.head()

,Date,HomeTeam,AwayTeam,FTHG,FTAG,FTR
0,24/08/2018,Bayern Munich,Hoffenheim,3,1,H
1,25/08/2018,Fortuna Dusseldorf,Augsburg,1,2,A
2,25/08/2018,Freiburg,Ein Frankfurt,0,2,A
3,25/08/2018,Hertha,Nurnberg,1,0,H
4,25/08/2018,M'gladbach,Leverkusen,2,0,H


## 5. First Descriptive Statistics

To obtain an initial overview of the match outcomes, we calculate the frequency of home wins, away wins, and draws.

These descriptive statistics provide a first indication of the distribution of results before comparing different seasons.

In [39]:
relevant_df["FTR"].value_counts()

FTR
H    138
A     95
D     73
Name: count, dtype: int64

In [40]:
relevant_df["FTR"].value_counts(normalize=True) * 100

FTR
H    45.098039
A    31.045752
D    23.856209
Name: proportion, dtype: float64

### Interpretation

The results show that home wins occur more frequently than away wins during the 2018/19 Bundesliga season. This observation supports the assumption of a home advantage and serves as a baseline for later comparisons with the COVID-19 seasons.